src/metpy/calc/thermo.py

```diff
@@ -3264,3 +3264,59 @@ def gradient_richardson_number(height, potential_temperature, u, v, vertical_dim
     dvdz = first_derivative(v, x=height, axis=vertical_dim)
 
     return (mpconsts.g / potential_temperature) * (dthetadz / (dudz ** 2 + dvdz ** 2))
+
+
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def showalter_index(pressure, temperature, dewpt):
+    """Calculate Showalter Index from pressure temperature and 850 hPa lcl.
+
+    Showalter Index derived from [Galway1956]_:
+    SI = T500 - Tp500
+
+    where:
+    T500 is the measured temperature at 500 hPa
+    Tp500 is the temperature of the lifted parcel at 500 hPa
+
+   Parameters
+   ----------
+
+        pressure : `pint.Quantity`
+            Atmospheric pressure level(s) of interest, in order from highest to
+        lowest pressure
+
+        temperature : `pint.Quantity`
+            Parcel temperature for corresponding pressure
+
+        dewpt (:class: `pint.Quantity`):
```

Comment:
For our doc style this should look like:
```suggestion
        dewpt : `pint.Quantity`
```